cmd/instance_show.go

```diff
@@ -93,106 +106,139 @@ func (c *instanceShowCmd) cmdRun(cmd *cobra.Command, _ []string) error {
 
 		return nil
 	}
+	var ipV6 *net.IP
+	if parsed := net.ParseIP(instance.Ipv6Address); parsed != nil {
+		ipV6 = &parsed // only assign pointer if it's a valid IP
+	}
+
+	var sshKeyName *string
+	if instance.SSHKey != nil {
+		sshKeyName = &instance.SSHKey.Name
+	}
 
 	out := instanceShowOutput{
 		AntiAffinityGroups: make([]string, 0),
-		CreationDate:       instance.CreatedAt.String(),
-		DiskSize:           humanize.IBytes(uint64(*instance.DiskSize << 30)),
+		CreationDate:       instance.CreatedAT.String(),
+		DiskSize:           humanize.IBytes(uint64(instance.DiskSize << 30)),
 		ElasticIPs:         make([]string, 0),
-		ID:                 *instance.ID,
-		IPAddress:          utils.DefaultIP(instance.PublicIPAddress, "-"),
-		IPv6Address:        utils.DefaultIP(instance.IPv6Address, "-"),
+		ID:                 instance.ID,
+		IPAddress:          utils.DefaultIP(&instance.PublicIP, "-"),
+		IPv6Address:        utils.DefaultIP(ipV6, "-"),
```

Comment:
Alternatively, no need to parse in advance:

```suggestion
		IPv6Address:        utils.DefaultString(&instance.Ipv6Address, "-"),

```

Reply:
```cannot use &instance.Ipv6Address (value of type *string) as *net.IP value in argument to utils.DefaultIP```

Reply:
the thing is that i get
```
┼──────────────────────┼──────────────────────────────────────┼
│   COMPUTE INSTANCE   │                                      │
┼──────────────────────┼──────────────────────────────────────┼

│ Private Instance     │ No                                   │
│ Private Networks     │ n/a                                  │
│ Elastic IPs          │ n/a                                  │
│ IP Address           │ 159.100.241.227                      │
│ IPv6 Address         │ <nil>                                │
│ SSH Key              │ -                                    │
│ Disk Size            │ 50 GiB                               │

│ Reverse DNS          │                                      │
┼──────────────────────┼──────────────────────────────────────┼
```